src/charm.py

```diff
@@ -1,117 +1,296 @@
-                    "environment": {
-                        "GUNICORN_CMD_ARGS": f"--log-level {self.model.config['log-level']}"
-                    },
+        hostname = event.params["hostname"]
+        tls_certificates_relation = self._tls.get_tls_relation()
+        tls_secret_name = get_config(self, "tls-secret-name")
+
+        if not tls_certificates_relation and not tls_secret_name:
+            event.fail("Certificates relation not created.")
+            return
+
+        if tls_secret_name:
+            event.fail(
+                "Getting certificate from charm configuration is not yet supported",
+            )
+            return
+
+        tls_rel_data = tls_certificates_relation.data[self.app]
+        if tls_rel_data.get(f"certificate-{hostname}"):
+            event.set_results(
+                {
+                    f"certificate-{hostname}": tls_rel_data[f"certificate-{hostname}"],
+                    f"ca-{hostname}": tls_rel_data[f"ca-{hostname}"],
+                    f"chain-{hostname}": tls_rel_data[f"chain-{hostname}"],
                 }
-            },
-        }
+            )
+        else:
+            event.fail("Certificate not available")
+
+    def _on_certificates_relation_created(self, event: RelationCreatedEvent) -> None:
+        """Handle the TLS Certificate relation created event.
+
+        Args:
+            event: The event that fires this method.
+        """
+        if not self._are_relations_ready(event):
+            return
+        hostname = self.get_hostname()
+        if not hostname:
+            self.unit.status = BlockedStatus("Waiting for hostname to be configured")
+            event.defer()
+            return
+        self._tls.certificate_relation_created(hostname)
+
+    def _on_certificates_relation_joined(self, event: RelationJoinedEvent) -> None:
+        """Handle the TLS Certificate relation joined event.
+
+        Args:
+            event: The event that fires this method.
+        """
+        if not self._are_relations_ready(event):
+            return
+        hostname = self.get_hostname()
+        if not hostname:
+            self.unit.status = BlockedStatus("Waiting for hostname to be configured")
+            event.defer()
+            return
+        self._tls.certificate_relation_joined(hostname, self.certificates)
+
+    def _on_certificates_relation_broken(self, _: Any) -> None:
+        """Handle the TLS Certificate relation broken event."""
+        self._reconcile()
+
+    def _on_certificate_available(self, event: CertificateAvailableEvent) -> None:
+        """Handle the TLS Certificate available event.
+
+        Args:
+            event: The event that fires this method.
+        """
+        tls_certificates_relation = self._tls.get_tls_relation()
+        if not tls_certificates_relation:
+            self.unit.status = WaitingStatus("Waiting for certificates relation to be created")
+            event.defer()
```

Comment:
Analogous case here

Reply:
I'm marking this as resolved as per my comment above.